### ISSUE TITLE
Add support for grouping convolution filters

### ISSUE THREAD

**Issue by mkaze** [JetBrains/KotlinDL#112]
Currently, the support for grouping convolution filters in convolution layers is missing. For more information, you can read the `groups` argument description of `Conv2D` layer in Keras [here](https://www.tensorflow.org/api_docs/python/tf/keras/layers/Conv2D#args), or in PyTorch [here](https://pytorch.org/docs/stable/generated/torch.nn.Conv2d.html#torch.nn.Conv2d).

**Comment by mkaze** [comment]
Though, currently the underlying TF Java API does not support grouped convolutions. Alternatively, this could be implemented by grouping the input channels, applying filters separately and then concatenating the results.

**Comment by zaleslaw** [comment]
Agree, that this functionality is missed, the initial idea to make MVP DL framework - to implement building blocks required to make parse and load CV models from the Keras.applications. As I understand, that groups are not used in the SOTA models, but  maybe you could share architecture, competition or abstract/paper where it was used. I mean, do we really need this functionality?

**Comment by mkaze** [comment]
@zaleslaw I think ResNeXt used grouped convolutions; but I agree that it's not a crucial feature to have at this moment, especially for an MVP.

**Comment by zaleslaw** [comment]
If it is used in ResNeXt - it could make sense, the problem with ResNeXt is the following: it's not on the official list of pre-trained Keras models yet, but the model [is created ](https://github.com/tensorflow/tensorflow/issues/38711)

**Comment by zaleslaw** [comment]
P.S Read the [Contributing Guidelines](https://github.com/JetBrains/KotlinDL/blob/master/CONTRIBUTING.md).
